src/classes/UTIL_Finder.cls

```diff
@@ -0,0 +1,360 @@
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2018
+* @group Utilities
+* @description The SObject Finder use to query, filter, sort and limit number of records by using SOQL or SOSL
+*/
+public with sharing class UTIL_Finder implements UTIL_QueryApi.Finder {
+    /** @description Search exception class */
+    public class SoslException extends Exception {}
+
+    /** @description Message to indicate the Search is constructed incorrectly when sObjectType is not provided */
+    @TestVIsible private static final String SOBJECT_TYPE_REQUIRED = 'sObjectType is required for the search';
+    /** @description Message to indicate the Search is constructed incorrectly when no value is specified in FIND clause */
+    @TestVIsible private static final String SEARCH_CRITERIA_REQUIRED = 'Search criteria is required for the search';
+    /** @description Message to indicate the Search is constructed incorrectly when no sObject field is specified */
+    @TestVIsible private static final String FIELDS_REQUIRED = 'At least one field required to be returned from the search';
+
+    /** @description The sObjectType to query/search */
+    private sObjectType sObjType;
+    /** @description Fields to return from the query/search */
+    @TestVisible private Set<String> selectFields {
+        get {
+            if (selectFields == null) {
+                selectFields = new Set<String>{ 'Id' };
+            }
+            return selectFields;
+        }
+        set;
+    }
+    /** @description The WHERE clause */
+    @TestVisible private UTIL_Where criteria;
+    /** @description The ORDER BY segment */
+    private UTIL_OrderBy orderBy;
+    /** @description The LIMIT segment */
+    private Integer limitRows;
+
+    /*******************************************************************************************************************
+    * @description THe constructor for the provided sObjectType
+    */
+    public UTIL_Finder(sObjectType sObjType) {
+        this.sObjType = sObjType;
+        this.orderBy = new UTIL_OrderBy();
+        this.criteria = new UTIL_Where();
+    }
+
+    /*********************************************************************************************************
+    * @description Sets SELECT fields in the SOQL or RETURNING fields in the SOSL
+    * @param sObjFields Fields
+    * @return UTIL_Finder The object instance
+    */
+    public UTIL_Finder withSelectFields(Set<Schema.sObjectField> sObjFields) {
```

Comment:
Might not be a bad idea to note this in the method description or param line

Reply:
withSelectFields(Set<Schema.sObjectField> sObjFields, Schema.SObjectField referenceField) method allows you to specify the relationship field in order to add parent fields.

Reply:
@jjbennett , I have removed the method since it is not used currently. When there is a need for the method, we can include it. Let me know if there is any concern.

Reply:
Ok, sorry I was looking at an older version then. I am not concerned with the change, we can add it if we need it. Thanks!